analyzers/src/SonarAnalyzer.VisualBasic/Rules/UseReturnStatement.cs

```diff
@@ -0,0 +1,92 @@
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.VisualBasic;
+using Microsoft.CodeAnalysis.VisualBasic.Syntax;
+using SonarAnalyzer.Common;
+using SonarAnalyzer.Helpers;
+
+namespace SonarAnalyzer.Rules.VisualBasic
+{
+    [DiagnosticAnalyzer(LanguageNames.VisualBasic)]
+    [Rule(DiagnosticId)]
+    public sealed class UseReturnStatement : SonarDiagnosticAnalyzer
+    {
+        private const string DiagnosticId = "S5944";
+        private const string MessageFormat = "Use a 'Return' statement; assigning returned values to function names is obsolete.";
+        private static readonly DiagnosticDescriptor Rule = DiagnosticDescriptorBuilder.GetDescriptor(DiagnosticId, MessageFormat, RspecStrings.ResourceManager);
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        protected override void Initialize(SonarAnalysisContext context) =>
+            context.RegisterSyntaxNodeActionInNonGenerated(c =>
+            {
+                var method = (MethodStatementSyntax)((MethodBlockSyntax)c.Node).BlockStatement;
+                new Identifierswalker(c, method.Identifier.ValueText).SafeVisit(c.Node);
+            },
+            SyntaxKind.FunctionBlock);
+
+        private class Identifierswalker : VisualBasicSyntaxWalker
+        {
+            private SyntaxNodeAnalysisContext Context { get; }
+            private string Name { get; }
+
+            public Identifierswalker(SyntaxNodeAnalysisContext context, string name)
+            {
+                Context = context;
+                Name = name;
+            }
+
+            public override void VisitIdentifierName(IdentifierNameSyntax node)
+            {
+                if (IsImplicitReturnValue(node))
+                {
+                    if (NoAssignment(node))
+                    {
+                        // TODO: find the best way to do this.
+                        var rule = DiagnosticDescriptorBuilder.GetDescriptor(DiagnosticId, "Do not make use of the implicit return value.", RspecStrings.ResourceManager);
+                        Context.ReportDiagnosticWhenActive(Diagnostic.Create(rule, node.GetLocation()));
+                    }
+                    else
+                    {
+                        Context.ReportDiagnosticWhenActive(Diagnostic.Create(Rule, node.GetLocation()));
+                    }
+                }
+            }
+
+            private static bool IsExcluded(SyntaxNode node) =>
+               node.Parent is InvocationExpressionSyntax
+               || node.Parent is MemberAccessExpressionSyntax
+               || node.Parent is NamedFieldInitializerSyntax;
+
+            private bool IsImplicitReturnValue(IdentifierNameSyntax node) =>
```

Comment:
I'd put `IsExcluded` behind IsImplicitReturnValue` in the code, to follow their logical order.

Reply:
The rules asked me to put static before non-static, so I made them all non-static.

Reply:
Please ignore that rule. We're currently dogfooding the StyleCop rules and we'll remove that one.

Reply:
...and make the method `static` as it was